src/group/managed_group/mod.rs

```diff
@@ -214,6 +213,30 @@ impl ManagedGroup {
 
     /// Persists the state
     pub fn save(&self, writer: Box<dyn Write>) {}
+
+    // === Interface for callbacks ===
+
+    /// Get group ID
+    pub fn group_id(&self) -> GroupId {
```

Comment:
Should this return a reference?
```suggestion
    pub fn group_id(&self) -> &GroupId {
```